ding/reward_model/count_based_model.py

```diff
@@ -0,0 +1,354 @@
+import torch
+from torch import distributions
+from torch import nn
+import torch.optim as optim
+import torch.nn.functional as F
+import cv2
+import random
+from math import log, exp, pow
+
+import numpy as np
+from ding.utils import REWARD_MODEL_REGISTRY
+from .base_reward_model import BaseRewardModel
+
+
+@REWARD_MODEL_REGISTRY.register('countbased')
+class CountbasedRewardModel(BaseRewardModel):
+    """
+        # TODO:
+    """
+    
+    config = dict(
+        type='countbased',
+        counter_type='PixelCNN',
+        intrinsic_reward_type='add',
+        # learning_rate=1e-3,
+        batch_size=16,
+        bonus_coeffient=0.1,
+        img_height=42,
+        img_width=42,
+        in_channels=1,
+        out_channels=1,
+        n_gated=2,
+        gated_channels=16,
+        head_channels=64,
+        q_level=256,
+    )
+
+    def __init__(
+        self,
+        cfg: dict,
+        device,
+        tb_logger: 'SummaryWriter'
+    ) -> None:  # noqa
+        """
+            # TODO:
+        """
+        super(BaseRewardModel, self).__init__()
+        self.cfg: dict = cfg
+        self._beta = cfg.bonus_coeffient
+        self._counter_type = cfg.counter_type
+        self.device = device
+        self.tb_logger = tb_logger
+        assert self._counter_type in ['SimHash', 'AutoEncoder', 'PixelCNN']
+        # TODO: copy
+        if self._counter_type == 'PixelCNN':
+            print(cfg)
+            self._counter = GatedPixelCNN(
+                in_channels=cfg.in_channels,
+                out_channels=cfg.out_channels,
+                n_gated=cfg.n_gated,
+                gated_channels=cfg.gated_channels,
+                head_channels=cfg.head_channels,
+                q_level=cfg.q_level,
+            )
+            self._counter.to(self.device)
+            self.intrinsic_reward_type = cfg.intrinsic_reward_type
+            self.obs_shape = (cfg.img_height, cfg.img_width)
+            self.index_range = np.arange(cfg.img_height * cfg.img_width)
+            assert self.intrinsic_reward_type in ['add', 'new', 'assign']
+            self.train_data = []
+            self.opt = optim.RMSprop(
+                self._counter.parameters(),
+                momentum=0.9,
+                weight_decay=0.95,
```

Comment:
Why use 0.95 weight_decay here, usually it is a small positive value (e.g.: 1e-3 or 1e-4)

Reply:
the paper uses decay=0.95,  it can be seen in page 11 of https://arxiv.org/pdf/1703.01310.pdf.